### PR TITLE
Render upload progress/completion public

### PR DESCRIPTION
**Pull request by SjoerdOptLand**
When using some JavaScript-based client-side uploader instead of locally calling upon the `upload()` function, still some tasks need to be done server-side: checking and completing the upload. 

Therefore, I factored out these two tasks from the `upload()` function into public methods `uploadedBytes()` and `completeUpload()`.

I hope these fit the naming conventions and general API philosophy. If not, please let me know.

### CODE REVIEW

File: vimeo.php [hunk 1]
```diff
@@ -326,6 +359,8 @@ public function upload ($file_path, $machine_id = null) {
     }
 }
 
+
+
```
Comment (gquemener) [inline vimeo.php]: This extra lines are not really needed :)

**Comment by gquemener** [comment]
Wouldn't using curl opt `CURLOPT_PROGRESSFUNCTION` add less overhead?

I mean, the solution you propose is to make more calls to the complete uri whereas curl ext already knows how many bytes he has transfered.

Here's a closure example with a Symfony ProgressBar instance:
``` php
        $progress = new ProgressBar($output, filesize($videoFilename));
        $closure = function ($resource, $downloadSize, $downloaded, $uploadSize, $uploaded) use ($progress) {
            $units = array('B', 'KB', 'MB', 'GB', 'TB');

            $bytes = curl_getinfo($resource, CURLINFO_SPEED_UPLOAD);
            $bytes = max($bytes, 0);
            $pow = floor(($bytes ? log($bytes) : 0) / log(1024));
            $pow = min($pow, count($units) - 1);

            $bytes /= (1 << (10 * $pow));
            $speed = round($bytes, 2) . ' ' . $units[$pow] . '/s';
            $progress->setMessage($speed, 'upload_speed');

            $progress->setCurrent($uploaded);
        };
```

**Comment by gquemener** [comment]
Anyway, could you provide an example to show progress status with a [Symfony ProgressBar](https://github.com/symfony/Console/blob/master/Helper/ProgressBar.php) by using the solution you propose in this PR, please?

**Comment by RugWarrior** [comment]
Hi @SjoerdOptLand may you have look at your PR?

**Review by pp1122122 — APPROVED** [review]

**Comment by tommypenner** [comment]
Closing this out as it's a few years old (!!) and the resumable upload method has been replaced with tus

**Pull request closed without merging**